homeassistant/core.py

```diff
@@ -2627,8 +2631,12 @@ async def async_call(
         if not return_response:
             return None
         if not isinstance(response_data, dict):
-            raise HomeAssistantError(
-                f"Service response data expected a dictionary, was {type(response_data)}"
+            raise ServiceValidationError(
```

Comment:
Changed back to `HomeAssistantError`. About the translation. We fetch the stings from translation cache for logging too. This error can happen in the UI, only in that case it will show a translation if it is available.